src/ngraph/ngraph_stats.h

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef MXNET_NGRAPH_NGRAPH_STATS_H
+#define MXNET_NGRAPH_NGRAPH_STATS_H
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "ngraph_graph.h"
+
+namespace ngraph_bridge {
+
+/// A singleton class to track and output nGraph performance statistics.
+class NGraphStats {
+ public:
+  static NGraphStats& get_instance() {
+    static NGraphStats instance;
+    return instance;
+  }
+
+  // disable copy
+  NGraphStats(NGraphStats const&) = delete;
+  void operator=(NGraphStats const&) = delete;
+
+  void add(const std::shared_ptr<Graph>& g) { graphs_.push_back(g); }
+  void dump(std::ostream& stream);
+
+ private:
+  // disallow creating instance outside the class
+  NGraphStats() {}
```

Comment:
Why does it need to be a singleton? I hate singletons...

Reply:
I wonder if we extend or use mxnet profiler singleton somehow , allows more seamless integration if so. `src/profiler/profiler.h:340`

Reply:
When I talk to @mbrookhart, he's still on the fence of whether this code will be dev only or going up stream. I made it a singleton not for generality but for simplicity (otherwise we have to consider its creation/life time and who owns it etc.). It's intended to be use once at the end of a model run for us to evaluate performance. If we find other important use cases that warrant us to invest more time/effort on this, we can always improve it further down the road.

Reply:
I'm minorly against the singleton use here, but it's definitely reasonable for you to ignore me on this one.